### ISSUE TITLE
No way to resample in browser-ui when optimization fails

### ISSUE THREAD

**Issue by keenancrane** [penrose/penrose#878]
**Describe the bug**

Currently if optimization fails, the `resample` button is disabled in the browser UI.  This is a problem, because re-running the same variant will of course again cause optimization to fail.  Currently the only solution is to manually type a new variant, and then refresh the browser.  This solution is clunky, and non-obvious to most users.

**Expected behavior**

The simplest fix is to just ensure that the `resample` button is always active.

Basically we want to let optimization "fail gracefully": even if we didn't reach a configuration that satisfies the optimality conditions, we can show the most recent configuration found by the optimizer—as well as a warning that optimization did not converge.  But we shouldn't prevent users from re-sampling in this case.
